javascript/sbp/observation.js

```diff
@@ -0,0 +1,1016 @@
+  .endianess('little')
+  .int32('i')
+  .uint8('f');
+
+var ObservationHeader = function (sbp, msg) {
+  
+  /*SBP class for message fragment ObservationHeader
+
+  
+  Header of a GPS observation message.
+
+  Parameters
+  ----------
+  t : ObsGPSTime
+    GPS time of this observation
+  n_obs : int
+    Total number of observations. First nibble is the size
+of the sequence (n), second nibble is the zero-indexed
+counter (ith packet of n)
+
+  sender : int
+    Optional sender ID, defaults to SENDER_ID (see sbp/msg.py).
+
+  */
+
+  this.sbp = sbp;
+  this.fields = this.parser.parse(msg);
+
+  return this;
+};
+ObservationHeader.prototype = Object.create(SBP.prototype);
+ObservationHeader.prototype.constructor = ObservationHeader;
+ObservationHeader.prototype.toBuffer = function () {
+  var buffers = [];
+  var b;
+  buffers = buffers.concat( ObsGPSTime.prototype.toBuffer.apply({ fields: this.fields.t }));
+  b = new Buffer( 1 );
+  b.writeUInt8(this.fields.n_obs);
+  buffers.push(b);
+  return Buffer.concat(buffers);
+};
+ObservationHeader.prototype.parser = new Parser()
+  .endianess('little')
+  .nest('t', { type: ObsGPSTime.prototype.parser })
+  .uint8('n_obs');
+
+var PackedObsContent = function (sbp, msg) {
+  
+  /*SBP class for message fragment PackedObsContent
+
+  
+  Pseudorange and carrier phase observation for a satellite being
+tracked.
+
+
+  Parameters
+  ----------
+  P : int
+    Pseudorange observation
+  L : CarrierPhase
+    Carrier phase observation
+  cn0 : int
+    Carrier-to-Noise density
+  lock : int
+    Lock indicator. This value changes whenever a satellite
+signal has lost and regained lock, indicating that the
+carrier phase ambiguity may have changed.
+
+  sid : int
+    Signal identifier of the satellite signal - values 0x00
+through 0x1F represent GPS PRNs 1 through 32 respectively
+(PRN-1 notation); other values reserved for future use.
+
+  sender : int
+    Optional sender ID, defaults to SENDER_ID (see sbp/msg.py).
+
+  */
+
+  this.sbp = sbp;
+  this.fields = this.parser.parse(msg);
+
+  return this;
+};
+PackedObsContent.prototype = Object.create(SBP.prototype);
+PackedObsContent.prototype.constructor = PackedObsContent;
+PackedObsContent.prototype.toBuffer = function () {
+  var buffers = [];
+  var b;
+  b = new Buffer( 4 );
+  b.writeUInt32LE(this.fields.P);
+  buffers.push(b);
+  buffers = buffers.concat( CarrierPhase.prototype.toBuffer.apply({ fields: this.fields.L }));
+  b = new Buffer( 1 );
+  b.writeUInt8(this.fields.cn0);
+  buffers.push(b);
+  b = new Buffer( 2 );
+  b.writeUInt16LE(this.fields.lock);
+  buffers.push(b);
+  b = new Buffer( 4 );
+  b.writeUInt32LE(this.fields.sid);
+  buffers.push(b);
+  return Buffer.concat(buffers);
+};
+PackedObsContent.prototype.parser = new Parser()
+  .endianess('little')
+  .uint32('P')
+  .nest('L', { type: CarrierPhase.prototype.parser })
+  .uint8('cn0')
+  .uint16('lock')
+  .uint32('sid');
+
+var MSG_OBS = function (sbp, msg) {
```

Comment:
Hrmmmm, what's both `sbp` and `msg` doing here? Is `sbp` a mixin?

You probably want to keep the original calling contract (using an existing sbp object or instantiating an object from a dict of fields):

```python
def __init__(self, sbp=None, **kwargs):
    if sbp:
      super( MsgTrackingIq,
             self).__init__(sbp.msg_type, sbp.sender, sbp.length,
                            sbp.payload, sbp.crc)
      self.from_binary(sbp.payload)
```